develop/wab/add_custom_theme/lab.md

```diff
@@ -0,0 +1,288 @@
+ 
+  ```json
+ {
+  "name": "HackerlabTheme",
+  "platform": "HTML",
+  "panels": [
+    {
+      "name": "FoldablePanel",
+      "description": "This is foldable panel"
+    }
+  ],
+  "styles": [
+    {
+      "name": "default",
+      "description": "this is default style",
+      "styleColor": "#2196f3"
+    },
+    {
+      "name": "medium-gray",
+      "description": "this is gray style",
+      "styleColor": "#6E6E6E"
+    },
+    {
+      "name": "black-gray",
+      "description": "this is black gray style",
+      "styleColor": "#2E3641"
+    },
+    {
+      "name": "white",
+      "description": "this is white with blue style",
+      "styleColor": "#FFFFFF"
+    }
+  ],
+  "layouts": [
+    {
+      "name": "default",
+      "description": "this is default layout"
+    }
+  ],
+  "widgets": [
+    {
+      "name": "HeaderController",
+      "description": "this is theme widget controller"
+    }
+  ],
+  "version": "2.1",
+  "wabVersion": "2.1",
+  "author": "Esri Inc",
+  "description": "HackerlabTheme",
+  "copyright": "",
+  "license": "http://www.apache.org/licenses/LICENSE-2.0"
+}
+  ```
+3. Launch Web AppBuilder using the startup batch file. Create a New Default (2D) app. Give it a title `Custom Theme` and click `OK`.
+
+4. Click the HackerlabTheme and click `Save`. Notice the default `Style` is light blue and contains one `Layout`, but the header controller is dark gray. You are going to style this theme using CSS.
+ 
+5. Click the `HackerlabTheme\styles\default` folder and edit the style.css. Make sure you save your changes after each step. Throughout this lab you will see the name jimu which is the [Jimu CSS framework] (https://developers.arcgis.com/web-appbuilder/api-reference/css-framework.htm) in Web AppBuilder. 
+
+      * Import a new font and change the background color. 
+
+       ```CSS
+         @import url(//fonts.googleapis.com/css?family=Ubuntu);
+        .jimu-main-font{
+          font-family: 'Ubuntu';
+        }
+        .jimu-main-background{
+          background-color: #fff;
+        }
+        ```
+      * Add a blue background color and opacity for the default widgets on the map.
+
+      ```CSS
+        .jimu-widget-homebutton,
+        .jimu-widget-mylocation,
+        .jimu-widget-zoomslider .zoom,
+        .jimu-widget-homebutton .HomeButton .home,
+        .jimu-widget-mylocation .place-holder{
+          background-color: #2196f3;
+          opacity: .9;
+        }
+      ```
+      * Change the title and subtitle colors in the header to gray. 
+
+      ```CSS       
+        .jimu-title,
+        .jimu-subtitle {
+          color: #323232;
+        }
+      ```
+      * Remove the shadow and add a border.
+    
+    ```CSS                  
```

Comment:
i'm sure you're sick of me by now, but this last .css code snippet doesn't have the same indentation as the others.